### ISSUE TITLE
argparse: default args in mutually exclusive groups

### ISSUE THREAD

**Issue by 7aa6e20b-8983-474f-b2ae-de7eff1caa04** [python/cpython#63143]
BPO | [18943](https://bugs.python.org/issue18943)
--- | :---
Nosy | @rhettinger, @pitrou, @wm75, @mlouielu
Files | <li>[test_argparse.diff](https://bugs.python.org/file31626/test_argparse.diff "Uploaded as text/plain at 2013-09-06.09:00:01 by @arigo"): Patch for test_argparse.py (for trunk, but also applies on 2.7 head)</li><li>[patch_1.diff](https://bugs.python.org/file31704/patch_1.diff "Uploaded as text/plain at 2013-09-09.22:38:22 by paul.j3")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2013-09-06.09:00:02.098>
labels = ['3.7', 'type-bug', 'library']
title = 'argparse: default args in mutually exclusive groups'
updated_at = <Date 2020-12-05.07:19:06.915>
user = 'https://github.com/arigo'
```

bugs.python.org fields:
```python
activity = <Date 2020-12-05.07:19:06.915>
actor = 'rhettinger'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2013-09-06.09:00:02.098>
creator = 'arigo'
dependencies = []
files = ['31626', '31704']
hgrepos = []
issue_num = 18943
keywords = ['patch']
message_count = 24.0
messages = ['197058', '197121', '197128', '197141', '197161', '197225', '197275', '197290', '197295', '197336', '197350', '197414', '197427', '211229', '212693', '292305', '292313', '293931', '299355', '307710', '307758', '307760', '307762', '307852']
nosy_count = 7.0
nosy_names = ['rhettinger', 'pitrou', 'bethard', 'paul.j3', 'wolma', 'louielu', 'talkless']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue18943'
versions = ['Python 2.7', 'Python 3.5', 'Python 3.7']
```

</p></details>

### Linked PRs
* gh-124307
* gh-124418
* gh-124419

**Comment by 7a064fe6-c535-4d80-a11f-a04ed39056c5** [comment]
This came up again, http://bugs.python.org/issue30163

An optional with int type and small integer default.

**Comment by 9da18a29-a07d-4828-a801-24cc911246de** [comment]
paul, will you work on this patch? or I can help this issue, too.

**Comment by 7a064fe6-c535-4d80-a11f-a04ed39056c5** [comment]
I haven't downloaded the development distribution to this computer, so can't write formal patches at this time.

**Comment by 7a064fe6-c535-4d80-a11f-a04ed39056c5** [comment]
Another manifestation of the complications in handling '?' positionals is in

http://bugs.python.org/issue28734

argparse: successive parsing wipes out nargs=? values

**Comment by serhiy-storchaka** [comment]
A simpler fix is included as the part of #124306. But since there is already specific issue and that PR do not have specific tests, I created a specific PR #124307.

Instead of relying on small integer cache, it uses bool for testing.
